setup/CLI/LiveConsoleProgress.cs

```diff
@@ -0,0 +1,102 @@
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.Timers;
+using Spectre.Console;
+using Terraria.ModLoader.Setup.Core.Abstractions;
+using Timer = System.Timers.Timer;
+
+namespace Terraria.ModLoader.Setup.CLI;
+
+public sealed class LiveConsoleProgress : IProgress, IDisposable
+{
+	private readonly LiveDisplayContext context;
+	private readonly Table table;
+	private readonly Timer timer;
+	private readonly ConcurrentDictionary<TaskProgress, object?> taskProgresses = [];
```

Comment:
Since the way we do tables for live console progress doesn't support two simultaneous progress updating tasks, it's probably better if the code doesn't pretend as such. It should be fine for this to just be `TaskProgress?` and the `onCompleted` just sets `taskProgress = null`. We can include an assertion that `taskProgress == null` in the constructor if we want to...

Alternatively, if we move the `timer` and `Refresh` methods into the `TaskProgress` class, we can `Dispose` it all together, and there's no need to track a reference here. (Similar to how the `MainForm` impl works)